docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -561,15 +561,19 @@ export async function generateStaticParams() {
 }
 ```
 
-To statically render all paths the first time they're visited, return an empty array (no paths will be rendered at build time):
+To statically render all paths the first time they're visited, return an empty array (no paths will be rendered at build time) or utilize [`export const dynamic = 'force-static'`](/docs/app/api-reference/file-conventions/route-segment-config#dynamic):
 
-```tsx filename="app/blog/[slug]/page.tsx" switcher
+```jsx filename="app/blog/[slug]/page.js"
 export async function generateStaticParams() {
   return []
 }
 ```
 
-> **Good to know:** You must always return an array from `generateStaticParams`, even if it's empty. Otherwise, the route will be dynamically rendered.
+```jsx filename="app/changelog/[slug]/page.js" switcher
+export const dynamic = 'force-static'
+```
+
+> **Good to know:** You must return an array from `generateStaticParams`, even if it's empty. Otherwise, the route will be dynamically rendered.
```

Comment:
This note only pertains to the `generateStaticParams` bit and is a bit contradictory if we're saying you can also use `force-static`